service/storage_proxy.cc

```diff
@@ -258,6 +259,28 @@ class shared_mutation : public mutation_holder {
     }
 };
 
+// shared mutation, but gets sent as a hint
+class hint_mutation : public shared_mutation {
+public:
+    using shared_mutation::shared_mutation;
+    virtual bool store_hint(db::hints::manager& hm, gms::inet_address ep, tracing::trace_state_ptr tr_state) override {
+        return false;
+    }
+    virtual future<> apply_locally(storage_proxy& sp, storage_proxy::clock_type::time_point timeout,
+            tracing::trace_state_ptr tr_state) override {
+        return make_exception_future<>(std::runtime_error("Executing hint locally doesn't make sense"));
+    }
+    virtual future<> apply_remotely(storage_proxy& sp, gms::inet_address ep, std::vector<gms::inet_address>&& forward,
+            storage_proxy::response_id_type response_id, storage_proxy::clock_type::time_point timeout,
+            tracing::trace_state_ptr tr_state) override {
+        tracing::trace(tr_state, "Sending a hint to /{}", ep);
+        auto& ms = netw::get_local_messaging_service();
+        return ms.send_hint_mutation(netw::messaging_service::msg_addr{ep, 0}, timeout, *_mutation,
```

Comment:
Why `cpu_id` is hard coded to `0`?

Reply:
I don't see where you send this on a separate (or streaming) socket.
Do I miss something?

Reply:
cpu_id is always ignored. It is fine to hardcode it to whatever.

Reply:
@vladzcloudius Relevant changes are situated in `messaging_service.cc`. The client for given request is chosen based on `s_rpc_client_idx_table` array, which is initialized based on `do_get_rpc_client_idx` function.

Reply:
> @vladzcloudius Relevant changes are situated in `messaging_service.cc`. The client for given request is chosen based on `s_rpc_client_idx_table` array, which is initialized based on `do_get_rpc_client_idx` function.

Oh, I see it now. Thanks.